logger/zap.go

```diff
@@ -0,0 +1,168 @@
+package logger
+
+import (
+	"log"
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type ZapLogger struct {
+	Zap *zap.SugaredLogger
+}
+
+var zapLogger *ZapLogger
+
+func init() {
+	logLevelEnv := os.Getenv("LOG_LEVEL")
+	var logLevel zapcore.Level
+	switch logLevelEnv {
+	case "DEBUG":
+		logLevel = zapcore.DebugLevel
+	case "INFO":
+		logLevel = zapcore.InfoLevel
+	case "WARN":
+		logLevel = zapcore.WarnLevel
+	case "ERROR":
+		logLevel = zapcore.ErrorLevel
+	case "DPANIC":
+		logLevel = zapcore.DPanicLevel
+	case "PANIC":
+		logLevel = zapcore.PanicLevel
+	case "FATAL":
+		logLevel = zapcore.FatalLevel
+	default:
+		logLevel = zapcore.DebugLevel
+	}
+	level := zap.NewAtomicLevel()
+	level.SetLevel(logLevel)
+	cfg := zap.Config{
+		Level:    level,
+		Encoding: "json",
+		EncoderConfig: zapcore.EncoderConfig{
+			TimeKey:        "TimeStamp",
+			LevelKey:       "Severity",
+			NameKey:        "Name",
+			CallerKey:      "Caller",
+			MessageKey:     "Body",
+			EncodeLevel:    zapcore.CapitalLevelEncoder,
+			EncodeTime:     zapcore.ISO8601TimeEncoder,
+			EncodeDuration: zapcore.StringDurationEncoder,
+			EncodeCaller:   zapcore.ShortCallerEncoder,
+		},
+		OutputPaths: []string{"stdout", "/tmp/zap.log"},
+	}
+	logger, err := cfg.Build(
+		zap.AddCallerSkip(2),
+	)
+	if err != nil {
+		log.Fatalf("CreateLogger Error: %v", err)
+	}
+
+	zapLogger = new(ZapLogger)
+	zapLogger.Zap = logger.Sugar()
+
+}
+
+func GetLogger() *ZapLogger {
+	return zapLogger
+}
+
+func (zl *ZapLogger) Errorw(
+	msg string,
+	err error,
+	stackKey string,
+) {
+	zl.Zap.Errorw(
+		msg,
+		"error", zap.Error(err),
+		"stack trace", zap.StackSkip(stackKey, 3),
+	)
```

Comment:
@dongrie Could you confirm this yourself to see if the following is correct?
```
zl.Zap.Errorw(
    msg,
    zap.Error(err),
    zap.StackSkip(stackKey, 3),
)
```

Please read the source of zap: https://github.com/uber-go/zap/blob/v1.24.0/sugar.go#L209-L211
If you feed a `Field` value to `Errorw`, `Infow`, and so on, you need not specify the key for the value.

Reply:
https://github.com/hyperledger-labs/yui-relayer/commit/34e3a9c77db7aab8a33b452c620ed63fc0b1cc7d
Fixed